plotter/barchart_test.go

```diff
@@ -0,0 +1,178 @@
+// Copyright ©2015 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package plotter
+
+import (
+	"image/color"
+	"testing"
+
+	"github.com/gonum/plot"
+	"github.com/gonum/plot/vg"
+)
+
+func ExampleBarChart() {
+	// Create the plot values and labels.
+	values := Values{0.5, 10, 20, 30}
+	verticalLabels := []string{"A", "B", "C", "D"}
+	horizontalLabels := []string{"Label A", "Label B", "Label C", "Label D"}
+
+	// Create a vertical BarChart
+	p1, err := plot.New()
+	handleEx(err)
```

Comment:
I did this because how we handle the error is not relevant to the example, so I didn't see any benefit from making the definition available to the reader. I see the benefit of doing it this way as allowing the reader to focus on the code that is relevant, whereas the potential disbenefit is that it could possibly cause confusion for the reason you point out. I will defer to your judgement regarding whether the benefits outweigh the drawbacks. A third option could be just to make the function name more descriptive or add a comment saying that the user can choose how to handle the error or something like that.